src/components/dac_dataset_table/DACDatasetApprovalStatus.js

```diff
@@ -1,14 +1,33 @@
 import React, {useState} from 'react';
-import {DAC} from '../../libs/ajax';
+import {DAC, DataSet} from '../../libs/ajax';
 import {Link} from 'react-router-dom';
 import {isNil} from 'lodash/fp';
 import Button from '@mui/material/Button';
 import ReactTooltip from 'react-tooltip';
 import style from '../../pages/DACDatasets.module.css';
+import { ConfirmationDialog } from '../modals/ConfirmationDialog';
+import IconButton from '@mui/material/IconButton';
+import DeleteForeverIcon from '@mui/icons-material/DeleteForever';
 
 export default function DACDatasetApprovalStatus(props) {
 
   const [dataset, setDataset] = useState(props.dataset);
+  const [open, setOpen] = useState(false);
+
+  const handleClick = () => {
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  const handleAction = (datasetId) => {
+    setOpen(false);
+    DataSet.deleteDataset(datasetId).then(() => {
+      window.location.reload();
+    });
+  }
```

Comment:
This function should only be called if the dataset is deletable. In the case that it isn't, the user will get an error:
```
{"message":"Dataset is in use and cannot be deleted.","code":400}
```
Look for the  `"deletable": false` property on the dataset to check this condition.

Reply:
I was not aware of this property! Good catch.